cmake/std/atdm/utils/get_known_system_info.sh

```diff
@@ -11,127 +11,196 @@ unset ATDM_CONFIG_CDASH_HOSTNAME
+  hostnameMatch=hansen
+  hostnameMatchSystemName=shiller
+elif [[ $realHostname == "shiller"* ]] ; then
+  hostnameMatch=shiller
+  hostnameMatchSystemName=shiller
+elif [[ $realHostname == "white"* ]] ; then
+  hostnameMatch=white
+  hostnameMatchSystemName=ride
+elif [[ $realHostname == "ride"* ]] ; then
+  hostnameMatch=ride
+  hostnameMatchSystemName=ride
+elif [[ $realHostname == "mutrino"* ]] ; then
+  hostnameMatch=mutrino
+  hostnameMatchSystemName=mutrino
+elif [[ $realHostname == "waterman"* ]] ; then
+  hostnameMatch=waterman
+  hostnameMatchSystemName=waterman
+
+# Specifically named cts1 systems
+elif [[ $realHostname == "serrano"* ]] \
+  || [[ $realHostname =~ ser[0-9]+ ]] ; then
+  hostnameMatch=serrano
+  hostnameMatchSystemName=
+elif [[ $realHostname == "eclipse"* ]] \
+  || [[ $realHostname =~ ec[0-9]+ ]] ; then
+  hostnameMatch=eclipse
+  hostnameMatchSystemName=serrano
+elif [[ $realHostname == "ghost"* ]] \
+  || [[ $realHostname =~ gho[0-9]+ ]] ; then
+  hostnameMatch=ghost
+  hostnameMatchSystemName=serrano
+
+# End specifically named systems
+fi
+
+#echo "hostnameMatch ='${hostnameMatch}'"
+
+#
+# C) Look for known system types that matches this machine
+#
+# A given machine may match more than one known system type so this can be an
+# array of matches.  Also, the list of known systems will be in the prefered
+# match order so, if no other match criteria is in play, then the first
+# matching system type will be selected.
+#
+
+systemNameTypeMatchedList=()  # In order of match preference
+unset systemNameTypeMatchedListHostNames
+declare -A systemNameTypeMatchedListHostNames
 
-# environments available on rhel systems
-elif [[ $ATDM_CONFIG_BUILD_NAME == *"spack-rhel"* ]] ; then
-  ATDM_HOSTNAME=spack-rhel
-  ATDM_SYSTEM_NAME=spack-rhel
-elif [[ -f /projects/sems/modulefiles/utils/get-platform ]] ; then
+# TLCC2 systems
+if [[ $SNLSYSTEM == "tlcc2"* ]] ; then
+  systemNameTypeMatchedList+=(tlcc2)
+  systemNameTypeMatchedListHostNames[tlcc2]=$SNLCLUSTER
+fi
+
+# SEMS and CEE RHEL systems
+if [[ -f /projects/sems/modulefiles/utils/get-platform ]] ; then
   # This machine has the SEMS modules!
-  ATDM_SYSTEM_NAME=`source /projects/sems/modulefiles/utils/get-platform`
-  if [[ $ATDM_SYSTEM_NAME == "rhel7-x86_64" ]] ; then
-    # This is a RHEL7 platform that has the SEMS modules.
-    ATDM_HOSTNAME=sems-rhel7
-    ATDM_SYSTEM_NAME=sems-rhel7
-  elif [[ $ATDM_SYSTEM_NAME == "rhel6-x86_64" ]] ; then
-    # This is a RHEL6 platform that has the SEMS modules.  But is this also a
-    # CEE LAN mahcine?
-    if [[ -f /projects/sparc/modules/cee-rhel6/sparc/master ]] ; then
-      ATDM_IS_CEE_RHEL6_MACHINE=1
+  ATDM_SEMS_SYSTEM_NAME=`source /projects/sems/modulefiles/utils/get-platform`
+  if [[ "${ATDM_SEMS_SYSTEM_NAME}" == "rhel6-x86_64" ]] ; then
+    systemNameTypeMatchedList+=(sems-rhel6)
+    systemNameTypeMatchedListHostNames[sems-rhel6]=sems-rhel6
+    if [[ -d /projects/sparc/tpls ]] ; then
```

Comment:
If this directory exists, we're on a cee-rhel6 machine?

Reply:
This is being fixed by using `SNLSYSTEM` and `SNLCLUSTER` as discussed in our meeting today.